systemtests/upgrade_test.go

```diff
@@ -0,0 +1,113 @@
+//go:build system_test && linux
```

Comment:
As the the old binary is fetched from docker, this does not run on mac. The hack for testing is to  compile the old binary on the system and copy it to the `binary` folder, so that it is not fetched again